src/main/java/sirius/biz/storage/layer2/BlobRefProperty.java

```diff
@@ -0,0 +1,126 @@
+
+    protected static final int DEFAULT_KEY_LENGTH = 64;
+
+    /**
+     * Creates a new property for the given descriptor, access path and field.
+     *
+     * @param descriptor the descriptor which owns the property
+     * @param accessPath the access path required to obtain the target object which contains the field
+     * @param field      the field which stores the database value
+     */
+    protected BlobRefProperty(@Nonnull EntityDescriptor descriptor,
+                              @Nonnull AccessPath accessPath,
+                              @Nonnull Field field) {
+        super(descriptor, accessPath, field);
+    }
+
+    /**
+     * Gets a {@link BlobHardRef} for the given entity.
+     *
+     * @param entity The entity to get the reference from
+     * @return a {@link BlobHardRef} or a subclass of it
+     */
+    protected BlobHardRef getRef(Object entity) {
+        try {
+            return (BlobHardRef) super.getValueFromField(this.accessPath.apply(entity));
+        } catch (Exception e) {
+            throw Exceptions.handle()
+                            .to(OMA.LOG)
+                            .error(e)
+                            .withSystemErrorMessage(
+                                    "Unable to obtain a reference object from entity ref field ('%s' in '%s'): %s (%s)",
+                                    getName(),
+                                    descriptor.getType().getName())
+                            .handle();
+        }
+    }
+
+    @Override
+    protected Object getValueFromField(Object target) {
+        return getRef(target).getKey();
+    }
+
+    @Override
+    public Object transformValue(Value value) {
+        return value.get();
+    }
+
+    @Override
+    protected Object transformToJDBC(Object object) {
+        return object;
+    }
+
+    @Override
+    protected Object transformFromJDBC(Value object) {
+        return object.get();
+    }
+
+    @Override
+    public void setValue(Object entity, Object object) {
+        this.setValueToField(object, entity);
+    }
+
+    /**
+     * Sets the given value on the target entity as either a blob or a key depending on the value.
+     *
+     * @param value        The value to be set
+     * @param targetEntity The entity to be set on
+     */
+    protected void setValueToField(Object value, Object targetEntity) {
+        BlobHardRef ref = getRef(targetEntity);
+
+        if (value == null || value instanceof Blob) {
+            ref.setBlob((Blob) value);
+        } else {
+            ref.setKey((String) value);
+        }
+    }
+
+    @Override
+    public void contributeToTable(Table table) {
+        table.getColumns().add(new TableColumn(this, Types.CHAR));
+    }
+
+    /**
+     * Checks if the blob reference on the entity has changed since loading from DB.
+     *
+     * @param entity the entity to check
+     * @return <tt>true</tt> if unchanged, else <tt>false</tt>
```

Comment:
```suggestion
     * @return <tt>true</tt> if changed, <tt>false</tt> otherwise
```

Reply:
the first part is the real change - second is just the common way to express the "else" part :)

Reply:
done